sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/InputsAdapter.java

```diff
@@ -249,14 +249,19 @@ public List<SclReportItem> updateAllSourceDataSetsAndControlBlocks() {
         return getExtRefs().stream()
             .filter(this::areBindingAttributesPresent)
             .filter(this::isExternalBound)
-            //TODO: replace filter below by criteria of issue #93 RSR-389
-            // - Active or Untested i.e /IED/LDevice/LN0/Inputs/Private/compas:Flow @FlowStatus="ACTIVE" OR "UNTESTED"
-            .filter(extRef -> !getMatchingCompasFlows(extRef, compasFlows).isEmpty())
+            .filter(extRef -> matchingCompasFlowIsActiveOrUntested(extRef, compasFlows))
             .map(extRef -> updateSourceDataSetsAndControlBlocks(extRef, currentBayUuid))
             .flatMap(Optional::stream)
             .toList();
     }
 
+    private boolean matchingCompasFlowIsActiveOrUntested(TExtRef extRef, List<TCompasFlow> compasFlows) {
+        return getMatchingCompasFlows(extRef, compasFlows).stream().findFirst()
```

Comment:
Next times it should be better to return only the stream and let the caller do the collect if necessary.
Collect is a terminal operation and is memory consuming

Reply:
`getMatchingCompasFlows` is used also by `updateExtRefIedName` where it is easier to work with a list. So I let the collect.